File: algorithms/2124.check-if-all-as-appears-before-all-bs/solution.rs
impl Solution {
    pub fn check_string(s: String) -> bool {
        let mut appeared = false;
        for c in s.chars() {
            match c {
                'a' => {
                    if appeared {
                        return false;
                    }
                }
                'b' => appeared = true,
                _ => {}
            }
        }

        true
    }
}

File: algorithms/2124.check-if-all-as-appears-before-all-bs/solution.rb
# @param {String} s
# @return {Boolean}
def check_string(s)
  appeared = false

  s.chars.each do |c|
    case c
    when 'a'
      return false if appeared
    when 'b'
      appeared = true
    end
  end

  true
end
